src/libpriv/rpmostree-core.cxx

```diff
@@ -623,13 +623,11 @@ rpmostree_context_get_rpmmd_repo_commit_metadata (RpmOstreeContext  *self)
   return g_variant_ref_sink (g_variant_builder_end (&repo_list_builder));
 }
 
-GHashTable *
+std::unique_ptr<rust::Vec<rpmostreecxx::StringMapping>> 
 rpmostree_dnfcontext_get_varsubsts (DnfContext *context)
```

Comment:
Hmm, can't we just oxidize this function too and basically inline it as part of `varsubstitute`? Then we avoid the map issue entirely.

Reply:
It depends on `DnfContext`...and we haven't yet crossed the bridge of using that with cxx-rs, which doing so sanely really requires a rework of the build system.